packages/components/src/dropdown-menu/index.js

```diff
@@ -59,13 +58,13 @@ function DropdownMenu( dropdownMenuProps ) {
 		noIcons,
 	} = dropdownMenuProps;
 
-	if ( isEmpty( controls ) && ! isFunction( children ) ) {
+	if ( ( ! controls || ! controls.length ) && ! isFunction( children ) ) {
```

Comment:
What about using optional chaining here? I think that would make it a little easier to parse.